### DISCUSSION TITLE
Acting as a Bluetooth LE Peripheral Device

### DISCUSSION THREAD

**Discussion by amrmarzouk** [orchetect/swift-midi#141] [Ideas]
I am relatively new here :) Any chance of adding MIDI Send events examples? :) 
Thanks a 10^6 !

**Comment by orchetect** [comment]
Bluetooth MIDI endpoints function the same as other endpoints in the system. If you establish a BT connection they will show up along side other endpoints and use the same API send and receive as demonstrated in the other example apps.

A test message send button could be added to the Bluetooth example apps but it would involve the user selecting the endpoint first from a list of endpoints available as you don't want to broadcast a message to all MIDI endpoints. I can add it to the to-do list and see if there's a simple way to add it to the demo.

**Reply by amrmarzouk** [reply]
That would be AWESOME ! Looking forward to it.
Thanks very much :)

**Reply by amrmarzouk** [reply]
Sorry I forgot to mention I am trying to send BLE MIDI messages as a Peripheral not Central device too.

**Reply by orchetect** [reply]
`CABTMIDILocalPeripheralViewController` may be what's needed. I'll dig a bit and see how much is involved in getting a proof of concept.

**Reply by orchetect** [reply]
Aha - got it all working. It's actually really easy. I'll update the Bluetooth demo projects and push them to the repo soon.

**Reply by orchetect** [reply]
Completed and pushed to new release: [0.7.0](https://github.com/orchetect/MIDIKit/releases/tag/0.7.0)

**Reply by amrmarzouk** [reply]
Awesome, you are a savior ! I was having a hard time making the jump from UIKit to SwiftUI with the `CABTMIDILocalPeripheralViewController` . I will give it a shot tomorrow !
Thanks :)
